src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -2835,6 +2835,76 @@ public override IOperation VisitCoalesce(ICoalesceOperation operation, int? capt
             return result;
         }
 
+        public override IOperation VisitCoalesceAssignment(ICoalesceAssignmentOperation operation, int? captureIdForResult)
+        {
+            SpillEvalStack();
+
+            // If we're in a statement context, we elide the capture of the result of the assignment, as it will
+            // just be wrapped in an expression statement that isn't used anywhere and isn't observed by anything.
+            bool isStatement = operation.Parent.Kind == OperationKind.ExpressionStatement;
+            Debug.Assert(captureIdForResult == null || !isStatement);
+
+            RegionBuilder resultCaptureRegion = _currentRegion;
+
+            EvalStackFrame frame = PushStackFrame();
+
+            IOperation testExpression;
+
+            PushOperand(Visit(operation.Target));
+            SpillEvalStack();
+            testExpression = PopOperand();
+
+            var whenNull = new BasicBlockBuilder(BasicBlockKind.Block);
+            var afterCoalesce = new BasicBlockBuilder(BasicBlockKind.Block);
+
+            int resultCaptureId = isStatement ? -1 : captureIdForResult ?? GetNextCaptureId(resultCaptureRegion);
+
+            LinkBlocks(CurrentBasicBlock,
+                       Operation.SetParentOperation(MakeIsNullOperation(testExpression), null),
+                       jumpIfTrue: true,
+                       RegularBranch(whenNull));
+
+            if (!isStatement)
+            {
+                _currentBasicBlock = null;
+
+                Debug.Assert(testExpression.Kind == OperationKind.FlowCaptureReference);
+                AddStatement(new FlowCapture(resultCaptureId, operation.Syntax, OperationCloner.CloneOperation(testExpression)));
+            }
+
+            LinkBlocks(CurrentBasicBlock, afterCoalesce);
+            _currentBasicBlock = null;
+
+            AppendNewBlock(whenNull);
+
+            // The return of Visit(operation.WhenNull) can be a flow capture that wasn't used in the non-null branch. We want to create a
+            // region around it to ensure that the scope of the flow capture is as narrow as possible. If there was no flow capture, region
+            // packing will take care of removing the empty region.
+            EvalStackFrame whenNullFrame = PushStackFrame();
+
+            IOperation whenNullValue = Visit(operation.Value);
+            IOperation nullAssignment = new SimpleAssignmentExpression(OperationCloner.CloneOperation(testExpression), isRef: false, whenNullValue, semanticModel: null,
```

Comment:
>nullAssignment [](start = 23, length = 14)

The name sounds like we're assigning `null`. Consider renaming. Perhaps `assignment` or `whenNullAssignment`.